### ISSUE TITLE
device-virtual got crash with autoevent error

### ISSUE THREAD

**Issue by cherrycl** [edgexfoundry/device-virtual-go#184]
Use the [docker-compose file](https://github.com/edgexfoundry/edgex-compose/blob/f2a853cccb20de9a336e390bf24c8b348df5b583/docker-compose-pre-release-no-secty.yml).

### Reproduce steps
1. Download the docker-compose file.
2. Start device-virtual with command `docker-compose -f docker-compose-pre-release-no-secty.yml up -d device-virtual`.
3. Remove the services with command `docker-compose -f docker-compose-pre-release-no-secty.yml down`. (Not remove volumes)
4. Start device-virtual again with command that the same as step2. 
5. Wait around 15s.

### Actual result
Found the following error in the logs.
```
level=ERROR ts=2021-04-15T06:58:36.011597475Z app=device-virtual source=transform.go:57 msg="failed to find DeviceResource  in Device Random-Integer-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:36.012676158Z app=device-virtual source=executor.go:54 msg="AutoEvent - error occurs when reading resource Int32: failed to convert CommandValue to Event -> failed to find DeviceResource  in Device Random-Integer-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:36.014429628Z app=device-virtual source=transform.go:57 msg="failed to find DeviceResource  in Device Random-Integer-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:36.014600112Z app=device-virtual source=executor.go:54 msg="AutoEvent - error occurs when reading resource Int64: failed to convert CommandValue to Event -> failed to find DeviceResource  in Device Random-Integer-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:36.01496817Z app=device-virtual source=transform.go:57 msg="failed to find DeviceResource  in Device Random-Integer-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:36.015017736Z app=device-virtual source=executor.go:54 msg="AutoEvent - error occurs when reading resource Int8: failed to convert CommandValue to Event -> failed to find DeviceResource  in Device Random-Integer-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:36.015236522Z app=device-virtual source=transform.go:57 msg="failed to find DeviceResource  in Device Random-Integer-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:36.015280606Z app=device-virtual source=executor.go:54 msg="AutoEvent - error occurs when reading resource Int16: failed to convert CommandValue to Event -> failed to find DeviceResource  in Device Random-Integer-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:41.011462127Z app=device-virtual source=transform.go:57 msg="failed to find DeviceResource  in Device Random-UnsignedInteger-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:41.011866567Z app=device-virtual source=executor.go:54 msg="AutoEvent - error occurs when reading resource Uint64: failed to convert CommandValue to Event -> failed to find DeviceResource  in Device Random-UnsignedInteger-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:41.013805575Z app=device-virtual source=transform.go:57 msg="failed to find DeviceResource  in Device Random-UnsignedInteger-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:41.014209486Z app=device-virtual source=executor.go:54 msg="AutoEvent - error occurs when reading resource Uint8: failed to convert CommandValue to Event -> failed to find DeviceResource  in Device Random-UnsignedInteger-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:41.014740878Z app=device-virtual source=transform.go:57 msg="failed to find DeviceResource  in Device Random-UnsignedInteger-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:41.014968511Z app=device-virtual source=executor.go:54 msg="AutoEvent - error occurs when reading resource Uint16: failed to convert CommandValue to Event -> failed to find DeviceResource  in Device Random-UnsignedInteger-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:41.015567692Z app=device-virtual source=transform.go:57 msg="failed to find DeviceResource  in Device Random-UnsignedInteger-Device for CommandValue (DeviceResource: , : <nil>)"
level=ERROR ts=2021-04-15T06:58:41.015898272Z app=device-virtual source=executor.go:54 msg="AutoEvent - error occurs when reading resource Uint32: failed to convert CommandValue to Event -> failed to find DeviceResource  in Device Random-UnsignedInteger-Device for CommandValue (DeviceResource: , : <nil>)"
panic: strconv: illegal AppendFloat/FormatFloat bitSize

goroutine 158 [running]:
strconv.genericFtoa(0xc000452220, 0x0, 0x18, 0x0, 0xc000096065, 0xffffffffffffffff, 0x0, 0x5350b1, 0xc000052230, 0x1675f5d3c5999a66)
	/usr/local/go/src/strconv/ftoa.go:68 +0x945
strconv.FormatFloat(0x0, 0x65, 0xffffffffffffffff, 0x0, 0xc000396218, 0x4)
	/usr/local/go/src/strconv/ftoa.go:48 +0xa7
github.com/edgexfoundry/device-virtual-go/internal/driver.(*resourceFloat).value(0x1022800, 0xc000020b80, 0xc0001f0a40, 0x13, 0xc0001f4c60, 0x7, 0x0, 0x0, 0x0, 0x0, ...)
	/device-virtual-go/internal/driver/resourcefloat.go:67 +0x31f
github.com/edgexfoundry/device-virtual-go/internal/driver.(*virtualDevice).read(0xc000358960, 0xc0001f0a40, 0x13, 0xc0001f4c60, 0x7, 0xc0001f4c67, 0x7, 0x0, 0x0, 0x0, ...)
	/device-virtual-go/internal/driver/virtualdevice.go:48 +0x4b0
github.com/edgexfoundry/device-virtual-go/internal/driver.(*VirtualDriver).HandleReadCommands(0xc000069590, 0xc0001f0a40, 0x13, 0xc0002d8270, 0xc0003e9200, 0x1, 0x1, 0xc0000112a8, 0x1, 0x1, ...)
	/device-virtual-go/internal/driver/virtualdriver.go:96 +0x2e5
github.com/edgexfoundry/device-sdk-go/v2/internal/application.(*CommandProcessor).ReadDeviceResource(0xc00047bb78, 0xc0001f0b60, 0x13, 0xc000191e70)
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.41/internal/application/command.go:143 +0x508
github.com/edgexfoundry/device-sdk-go/v2/internal/application.CommandHandler(0xad0001, 0x0, 0x0, 0xc0003f7d58, 0x0, 0x0, 0x0, 0x0, 0xc000299500, 0x0, ...)
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.41/internal/application/command.go:104 +0x57e
github.com/edgexfoundry/device-sdk-go/v2/internal/autoevent.readResource(0xc000068d20, 0xc000299500, 0x16, 0xc00042d820, 0x1)
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.41/internal/autoevent/executor.go:86 +0x1a5
github.com/edgexfoundry/device-sdk-go/v2/internal/autoevent.(*Executor).Run(0xc000068d20, 0xc6d2c0, 0xc0002b2b80, 0xc00028cb70, 0xc0003aee00, 0xc000299500)
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.41/internal/autoevent/executor.go:52 +0x2c5
created by github.com/edgexfoundry/device-sdk-go/v2/internal/autoevent.(*manager).triggerExecutors
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.41/internal/autoevent/manager.go:78 +0x273
```

**Comment by hahattan** [comment]
this could possibly be resolved by #185, requires verification ...

**Comment by cherrycl** [comment]
This issue does not happen on the latest image.
